plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/ui/internal/CDK3ServerWizardFragment.java

```diff
@@ -101,25 +101,27 @@ public void widgetDefaultSelected(SelectionEvent e) {
 
 	@Override
 	protected String findError() {
+		// Validate credentials
 		if( credentials.getDomain() == null || credentials.getUser() == null) {
 			return "The Container Development Environment Server Adapter requires Red Hat Access credentials.";
 		}
+		// Validate hypervisor
 		if( selectedHypervisor == null ) {
 			return "You must choose a hypervisor.";
 		}
 
-		
+		// Validate home directory
 		String retString = null;
-		String tmpStr = null;
 		if( homeDir == null || !(new File(homeDir)).exists()) {
-			tmpStr = "The selected file does not exist.";
+			retString = "The selected file does not exist.";
 		} else if( !(new File(homeDir).canExecute())) {
-			tmpStr = "The selected file is not executable.";
+			retString = "The selected file is not executable.";
 		}
+		toggleDecorator(homeText, retString);
```

Comment:
wondering why you'd pass homeText in as parameter since this method is hard coded on homeDecorator anyhow